core/client/src/main/java/alluxio/client/file/policy/DeterministicHashPolicy.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.client.file.policy.options.BlockLocationPolicyGetWorkerOptions;
+import alluxio.wire.WorkerNetAddress;
+
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Random;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * This policy maps blockId to several deterministic Alluxio workers. The number of workers a block
+ * can be mapped to can be passed through the constructor. The default is 1. It skips the workers
+ * that do not have enough capacity to hold the block.
+ */
+@NotThreadSafe
+public final class DeterministicHashPolicy implements BlockLocationPolicy {
+  /** The default number of shards to serve a block. */
+  private static final int DEFAULT_NUM_SHARDS = 1;
```

Comment:
The num shards is configurable. Why do we need to make the default configurable?